custom_components/tuya_local/devices/gosund_SW2_dimmer_switch.yaml

```diff
@@ -0,0 +1,28 @@
+name: Gosund SW2 Dimmer Switch
+products:
+  - id: qcgkaqmaivuwfwz4
+    manufacturer: Gosund
+    model: SW2-AR
+entities:
+  - entity: light
+    dps:
+      - id: 1
+        type: boolean
+        name: switch
+      - id: 3
+        type: integer
+        name: brightness
+        range:
+          min: 25
+          max: 255
+  - entity: binary_sensor
+    name: Temperature Alarm
```

Comment:
Naming: HA naming convention is to only capitalise the first word of a device or entity name.

Also, the class may be better as `heat` rather than the more generic `problem`